src/main/java/de/tum/cit/aet/artemis/programming/service/ProgrammingExerciseService.java

```diff
@@ -1072,4 +1079,45 @@ public ProgrammingExercise loadProgrammingExerciseWithAuxiliaryRepositories(long
         final Set<ProgrammingExerciseRepository.ProgrammingExerciseFetchOptions> fetchOptions = Set.of(AuxiliaryRepositories);
         return programmingExerciseRepository.findByIdWithDynamicFetchElseThrow(exerciseId, fetchOptions);
     }
+
+    /**
+     * Validates the network access feature for the given programming language.
+     * Currently, SWIFT and HASKELL do not support disabling the network access feature.
+     *
+     * @param programmingExercise the programming exercise to validate
+     */
+    public void validateDockerFlags(ProgrammingExercise programmingExercise) {
+        ProgrammingExerciseBuildConfig buildConfig = programmingExercise.getBuildConfig();
+
+        List<List<String>> flags = programmingExerciseBuildConfigService.parseDockerFlags(buildConfig);
+
+        if (flags != null) {
+            String envFlag = getEnvFlag(flags);
+
+            if (envFlag != null && envFlag.length() > MAX_ENVIRONMENT_VARIABLES_DOCKER_FLAG_LENGTH) {
+                throw new BadRequestAlertException("The environment variables are too long. Max " + MAX_ENVIRONMENT_VARIABLES_DOCKER_FLAG_LENGTH + " chars", "Exercise",
+                        "envVariablesTooLong");
+            }
+        }
+
+        DockerRunConfig dockerRunConfig = programmingExerciseBuildConfigService.getDockerRunConfigFromParsedList(flags);
+
+        if (dockerRunConfig == null) {
+            return;
+        }
+
+        if (List.of(ProgrammingLanguage.SWIFT, ProgrammingLanguage.HASKELL).contains(programmingExercise.getProgrammingLanguage()) && dockerRunConfig.isNetworkDisabled()) {
+            throw new BadRequestAlertException("This programming language does not support disabling the network access feature", "Exercise", "networkAccessNotSupported");
```

Comment:
It seems like a translation for `networkAccessNotSupported` is missing on the client side